archive/mozilla_mar.ksy

```diff
@@ -0,0 +1,116 @@
+meta:
+  id: mozilla_mar
+  title: Mozilla ARchive
+  file-extension: mar
+  license: CC0-1.0
+  endian: be
+doc: |
+  Mozilla ARchive file is Mozilla's own archive format to distribute software updates.
+  Test files can be found on Mozilla's FTP site, for example:
+
+  <http://ftp.mozilla.org/pub/firefox/nightly/partials/>
+doc-ref: https://wiki.mozilla.org/Software_Update:MAR
+seq:
+  - id: magic
+    contents: "MAR1"
+  - id: ofs_index
+    -orig-id: OffsetToIndex
+    type: u4
+  - id: file_size
+    -orig-id: FileSize
+    type: u8
+  - id: len_signatures
+    -orig-id: NumSignatures
+    type: u4
+  - id: signatures
+    type: signature
+    repeat: expr
+    repeat-expr: len_signatures
+  - id: len_additional_sections
+    -orig-id: NumAdditionalSections
+    type: u4
+  - id: additional_sections
+    type: additional_section
+    repeat: expr
+    repeat-expr: len_additional_sections
+instances:
+  index:
+    pos: ofs_index
+    type: mar_index
+types:
+  signature:
+    seq:
+      - id: algorithm
+        -orig-id: SignatureAlgorithmID
+        type: u4
+        enum: signature_algorithms
+      - id: len_signature
+        -orig-id: SignatureSize
+        type: u4
+      - id: signature
+        size: len_signature
+  additional_section:
+    seq:
+      - id: len_block
+        type: u4
+      - id: block_identifier
+        -orig-id: BlockIdentifier
+        type: u4
+        enum: block_identifiers
+      - id: bytes
+        size: len_block - len_block._sizeof - block_identifier._sizeof
+        type:
+          switch-on: block_identifier
+          cases:
+            block_identifiers::product_information: product_information_block
+  mar_index:
+    seq:
+      - id: len_index
+        -orig-id: IndexSize
+        type: u4
+      - id: index_entries
+        type: index_entries
+        size: len_index
```

Comment:
It's weird to have a field `len_index` inside a type `mar_index`. Just from the naming, I would assume that the `len_index` apparently means the size of the `mar_index` structure (so the byte size of `index_entries` field in this type would be `size: len_index - len_index._sizeof`). However, this is not the case and `len_index` is used as the size of `index_entries` substream, so why is `len_index` not called `len_index_entries`? If you strictly followed the naming convention of `len_` meaning **size** and `num_` for **number / count**, there would be no ambiguity.

```suggestion
  mar_index:
    seq:
      - id: len_index_entries
        -orig-id: IndexSize
        type: u4
      - id: index_entries
        type: index_entries
        size: len_index_entries
```